### PR TITLE
Enhance examples with outputs and images

### PR DESCRIPTION
**Pull request by hjmjohnson**
This is the next major set of changes to migrate the wiki examples to the sphinx examples.

### CODE REVIEW

**Comment by thewtex** [comment]
@hjmjohnson wow, amazing!!

LGTM after renaming the content links from `*.sha512.txt` to `*.sha512`.

**Comment by jhlegarreta** [comment]
@hjmjohnson @mseng10 :100: This was a very necessary process that will benefit the community as whole, those starting to learn ITK, advanced users, and maintainers ! **Thanks** for this huge effort.

**Comment by jhlegarreta** [comment]
Once merged, I guess it is reasonable to delete the [`ITKWikiExamplesRepositoryImport.todo`](https://github.com/InsightSoftwareConsortium/ITKExamples/blob/master/ITKWikiExamplesRepositoryImport.todo) file, and may be [`ITKRepositoryImport.todo`](https://github.com/InsightSoftwareConsortium/ITKExamples/blob/master/ITKRepositoryImport.todo) will need a review in case some of the examples have been covered by the migration. Thanks again.

File: src/Core/Common/CustomOperationToEachPixelInImage/Code.cxx [hunk 1]
```diff
@@ -20,7 +20,7 @@
 #include "itkVariableLengthVector.h"
 #include "itkRigid2DTransform.h"
 #include "itkUnaryFunctorImageFilter.h"
-#include "vnl/vnl_math.h"
+#include "itkMath.h"
```
Comment (dzenanz) [inline src/Core/Common/CustomOperationToEachPixelInImage/Code.cxx]: > requirements as vnl_math_ has been recently replaced with a namespace of vnl_math::

Commit message needs changing. After reading it, I though changes would now use `vnl_math::`. Examples now use `itk::Math`, not `vnl_math::`.

File: src/Developer/MultiThreadedImageFilter.h [hunk 2]
```diff
@@ -24,10 +24,10 @@ class MultiThreadedImageFilter : public ImageToImageFilter< TImage, TImage >
 
 protected:
   MultiThreadedImageFilter(){}
-  ~MultiThreadedImageFilter(){}
+  ~MultiThreadedImageFilter() override{}
 
-  virtual void ThreadedGenerateData(const OutputImageRegionType &,
-                                    ThreadIdType);
+  void ThreadedGenerateData(const OutputImageRegionType &,
```
Comment (dzenanz) [inline src/Developer/MultiThreadedImageFilter.h]: I guess we should rewrite this example to demonstrate the use of `DynamicThreadedGenerateData`. I created an issue #113 to track this.

File: src/Core/Common/CastVectorImageToAnotherType/Documentation.rst [hunk 3]
```diff
@@ -13,11 +13,9 @@ Cast a VectorImage to another type of VectorImage.
...
 Results
 -------
-.. note::
-   **Help Wanted**
-   Implementation of Results for sphinx examples containing this message.
-   Reconfiguration of CMakeList.txt may be necessary.
-   `Write An Example <https://itk.org/ITKExamples/Documentation/Contribute/WriteANewExample.html`>
+.. warning::
+  **Fix Errors**
+  Example contains errors needed to be fixed for proper output.
```
[the remark this answers is not in the document]
Comment (mseng10) [reply src/Core/Common/CastVectorImageToAnotherType/Documentation.rst]: Yes, the example contained errors not mentioned from the wiki so another pass will be necessary.

File: src/Core/Common/DirectWarningToFile/Documentation.rst [hunk 4]
```diff
@@ -13,11 +13,9 @@ Direct itk warnings to a file.
...
 Results
 -------
-.. note::
-   **Help Wanted**
-   Implementation of Results for sphinx examples containing this message.
-   Reconfiguration of CMakeList.txt may be necessary.
-   `Write An Example <https://itk.org/ITKExamples/Documentation/Contribute/WriteANewExample.html`>
+Output::
+
+  Look in itkMessageLog.txt for the output
```
[the remark this answers is not in the document]
Comment (mseng10) [reply src/Core/Common/DirectWarningToFile/Documentation.rst]: It should be included, I will go back and fix this.

File: src/Filtering/ImageFilterBase/ApplyKernelToEveryPixel/Documentation.rst [hunk 5]
```diff
@@ -13,11 +13,17 @@ Apply a kernel to every pixel in an image.
 
 Results
 -------
-.. note::
-   **Help Wanted**
-   Implementation of Results for sphinx examples containing this message.
-   Reconfiguration of CMakeList.txt may be necessary.
-   `Write An Example <https://itk.org/ITKExamples/Documentation/Contribute/WriteANewExample.html`>
+.. figure:: input.png
+  :scale: 70%
```
[the remark this answers is not in the document]
Comment (mseng10) [reply src/Filtering/ImageFilterBase/ApplyKernelToEveryPixel/Documentation.rst]: When reviewing already completed examples, some had a 70% value, thus I used it as a placeholder for the rest. I can go back and change to a different placeholder if needed.

File: src/Filtering/MathematicalMorphology/GenerateStructureElementsWithAccurateArea/Documentation.rst [hunk 6]
```diff
@@ -13,11 +13,112 @@ Generate structuring elements with accurate area.
...
+  0	  0	  0	  1	  1	  1	  1	  1	  0	  0	  0
+  Expected foreground area: 78.5398
+  Computed foreground area: 97
+  Foreground area error: 23.5042%
+
+  2D ball of radius 5 with radiusIsParametric mode on:
+  0	  0	  0	  0	  0	  1	  0	  0	  0	  0	  0
+  0	  0	  1	  1	  1	  1	  1	  1	  1	  0	  0
+  0	  1	  1	  1	  1	  1	  1	  1	  1	  1	  0
+  0	  1	  1	  1	  1	  1	  1	  1   1	  1	  0
```
[the remark this answers is not in the document]
Comment (mseng10) [reply src/Filtering/MathematicalMorphology/GenerateStructureElementsWithAccurateArea/Documentation.rst]: The misaligned errors are most likely due to copy and paste issues, however my window does not show the same output as the image above. Do you suggest re-entering output?
Screen Shot 2019-06-21 at 10 47 25 AM

File: src/Filtering/ImageFilterBase/ApplyKernelToEveryPixel/Documentation.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (dzenanz) [reply src/Filtering/ImageFilterBase/ApplyKernelToEveryPixel/Documentation.rst]: 70% is fine as a placeholder. The point of this is to make the small images larger, and large images smaller. Aim for perhaps 1000-1500 millimeters width. And the scale need to be adjusted per example/image. For example, make input1 and output1 both have 50% scale, while input2 could have 300% scale (hypothetical example). Again, depending on the number of pixels in the image and the spacing.

File: src/Filtering/MathematicalMorphology/GenerateStructureElementsWithAccurateArea/Documentation.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (dzenanz) [reply src/Filtering/MathematicalMorphology/GenerateStructureElementsWithAccurateArea/Documentation.rst]: You might have mixed tabs and spaces? Prefer all spaces. Some text editors have an option to convert all tabs to spaces.

**Pull request merged**